### PR TITLE
[AT-2835] Fixes wrt changes in product schema

### PR DESCRIPTION
**Pull request by seedlit**
Jira Ticket [here](https://up42.atlassian.net/browse/AT-2835).
Scope: Some changes were made to Product schema. 
 - This PR updates the functions to reflect those changes.
 - Updated tests
 - Updated mock schema
 - Added live tests

### CODE REVIEW

File: up42/utils.py [hunk 1]
```diff
@@ -366,16 +366,19 @@ def autocomplete_order_parameters(order_parameters: dict, schema: dict):
...
+                continue
+            elif "allOf" in schema["properties"][param]:  # has further definitions key
+                potential_values = [x["const"] for x in schema["definitions"][param]["anyOf"]]
+                logger.info(f"As `{param}` select one of {potential_values}")
+            else:
+                # Full information for simple parameters
+                del schema["properties"][param]["title"]
+                logger.info(f"As `{param}` select `{schema['properties'][param]}`")
+    except KeyError as exc:
+        raise KeyError("Please reach out to UP42 Support (support@up42.com)") from exc
```
Comment (Geosynopsis) [inline up42/utils.py:381]: Assuming this appears in many places through out library. Converting it to a constant somewhere is better perhaps

File: up42/utils.py [hunk 2]
```diff
@@ -366,16 +366,19 @@ def autocomplete_order_parameters(order_parameters: dict, schema: dict):
...
-        else:
-            # Full information for simple parameters
-            del schema["properties"][param]["title"]
-            logger.info(f"As `{param}` select `{schema['properties'][param]}`")
+    try:
+        for param in additional_params.keys():
+            if param in ["aoi", "geometry"]:
+                continue
+            elif "allOf" in schema["properties"][param]:  # has further definitions key
+                potential_values = [x["const"] for x in schema["definitions"][param]["anyOf"]]
```
Comment (Geosynopsis) [inline up42/utils.py:374]: From JSON schema's point of view, the ["anyOf"](https://json-schema.org/understanding-json-schema/reference/combining.html#anyof) could mave many possible configs. Is tasking guarantees that it won't be anything else?

File: up42/utils.py (hunk 1 above)
Comment (seedlit) [reply up42/utils.py:381]: Abhi, there are endless improvements that can be made. Idea is to minimally fix this function, and release a version to unblock everything else.

File: up42/utils.py (hunk 2 above)
Comment (seedlit) [reply up42/utils.py:374]: There is no such guarantee.
This entire function will be updated. See [this ticket](https://up42.atlassian.net/browse/INT-198).

File: up42/utils.py [hunk 3]
```diff
@@ -366,16 +366,19 @@ def autocomplete_order_parameters(order_parameters: dict, schema: dict):
...
+                logger.info(f"As `{param}` select one of {potential_values}")
+            else:
+                # Full information for simple parameters
+                del schema["properties"][param]["title"]
```
[the remark this answers is not in the document]
Comment (seedlit) [reply up42/utils.py:378]: Unrelated to this PR. Will skip this for the moment.
Like I mentioned, this entire function will be updated later :)

**Review by Geosynopsis — APPROVED** [review]

**Pull request merged**